source/extensions/filters/http/jwt_authn/filter.h

```diff
@@ -36,13 +39,14 @@ class Filter : public Http::StreamDecoderFilter,
   Http::StreamDecoderFilterCallbacks* decoder_callbacks_;
   // The stats object.
   JwtAuthnFilterStats& stats_;
-  // The auth object.
-  AuthenticatorPtr auth_;
   // The state of the request
   enum State { Init, Calling, Responded, Complete };
   State state_ = Init;
   // Mark if request has been stopped.
   bool stopped_ = false;
+  std::vector<AsyncMatcherSharedPtr> rule_matchers_;
+  Thread::MutexBasicLockable lock_;
```

Comment:
No need locking.  In Envoy,  all operations are done in the same thread.

Reply:
Even the async http client responses?